src/es_archiver/lib/indices/__tests__/create_index_stream.ts

```diff
@@ -67,14 +70,19 @@ describe('esArchiver: createCreateIndexStream()', () => {
           createStubIndexRecord('existing-index'),
           createStubIndexRecord('new-index'),
         ]),
-        createCreateIndexStream({ client, stats, log: { debug: () => {} } }),
+        createCreateIndexStream({ client, stats, log }),
       ]);
 
-      expect(client.indices.getAlias.calledOnce).to.be.ok();
-      expect(client.indices.getAlias.args[0][0]).to.eql({ name: 'existing-index', ignore: [404] });
-      expect(client.indices.delete.calledOnce).to.be.ok();
-      expect(client.indices.delete.args[0][0]).to.eql({ index: ['actual-index'] });
-      sinon.assert.callCount(client.indices.create, 3); // one failed create because of existing
+      expect((client.indices.getAlias as sinon.SinonSpy).calledOnce).to.be.ok();
+      expect((client.indices.getAlias as sinon.SinonSpy).args[0][0]).to.eql({
+        name: 'existing-index',
+        ignore: [404],
+      });
+      expect((client.indices.delete as sinon.SinonSpy).calledOnce).to.be.ok();
+      expect((client.indices.delete as sinon.SinonSpy).args[0][0]).to.eql({
```

Comment:
This is ugly. But `sinon` is way older than `jest`, and their mocking types are not as efficient. Had to keep concrete types for the stubs, so I need to hard cast on every sinon-related check.